web/ui/src/pages/PageComponentList.tsx

```diff
@@ -5,11 +5,32 @@ import Page from '../features/layout/Page';
 import { useComponentInfo } from '../hooks/componentInfo';
 
 function PageComponentList() {
-  const components = useComponentInfo('');
+  const [components, setComponents] = useComponentInfo('');
+
+  // TODO: make this sorting logic reusable
+  const handleSorting = (sortField: string, sortOrder: string): void => {
+    if (sortField) {
+      const sorted = [...components].sort((a, b) => {
+        const sortValueA = sortField === 'Health' ? a.health.state.toString() : a.localID;
+        const sortValueB = sortField === 'Health' ? b.health.state.toString() : b.localID;
+        if (sortValueA === null) return 1;
+        if (sortValueB === null) return -1;
+        if (sortValueA === null && sortValueB == null) return 0;
+        return (
+          sortValueA.localeCompare(sortValueB, 'en', {
+            numeric: true,
+          }) * (sortOrder === 'asc' ? 1 : -1)
+        );
+      });
+      if (sortOrder !== null) {
```

Comment:
maybe you could have the check also at the beginning. so it woud then be if (!sortfield || !sortOrder) return;